sdk/python/kfp/dsl/ontology_artifacts.py

```diff
@@ -54,13 +54,167 @@ def __init__(self):
 
   @classmethod
   def get_artifact_type(cls) -> str:
-    schema_file_path=os.path.join(
-          os.path.dirname(__file__), 'type_schemas', 'dataset.yaml')
+    return artifact_utils.read_schema_file('dataset.yaml')
+
+  @classmethod
+  def get_ir_type(cls) -> pipeline_spec_pb2.ArtifactTypeSchema:
+    return pipeline_spec_pb2.ArtifactTypeSchema(
+        instance_schema=cls.get_artifact_type())
+
+class Metrics(artifact.Artifact):
 
-    with open(schema_file_path) as schema_file:
-      return schema_file.read()
+  TYPE_NAME = 'kfp.Metrics'
+
+  def __init__(self):
+    self._instance_schema = self.__class__.get_artifact_type()
+
+    # Calling base class init to setup the instance.
+    super().__init__()
+
+  @classmethod
+  def get_artifact_type(cls) -> str:
+    return artifact_utils.read_schema_file('metrics.yaml')
+
+  @classmethod
+  def get_ir_type(cls) -> pipeline_spec_pb2.ArtifactTypeSchema:
+    return pipeline_spec_pb2.ArtifactTypeSchema(
+      instance_schema=cls.get_artifact_type())
+
+  def log_metric(self, metric: str, value: float):
+    self.metadata[metric] = value
+
+class ClassificationMetrics(artifact.Artifact):
```

Comment:
Yes, these are two different types of Metrics. Metrics == ScalarMetrics. We wanted to keep the name as Metrics because in MP it gets also used for RunComparison and having it named as Metrics is more readable.

Reply:
Makes sense. Thanks!